src/webgpu/gpu_test.ts

```diff
@@ -29,15 +35,35 @@ type TypedArrayBufferViewConstructor =
   | Float32ArrayConstructor
   | Float64ArrayConstructor;
 
+interface DeviceHolder {
+  device: GPUDevice;
+  lostReason?: string;
+}
+
+const kPopErrorScopeTimeoutMS = 5000;
+
 class DevicePool {
-  device: GPUDevice | undefined = undefined;
-  state: 'free' | 'acquired' | 'uninitialized' | 'failed' = 'uninitialized';
+  holder?: DeviceHolder = undefined; // undefined means "uninitialized"
+  state: 'free' | 'acquired' | 'failed' = 'free';
 
   private async initialize(): Promise<void> {
+    if (this.holder !== undefined) {
+      return;
+    }
     try {
       const gpu = getGPU();
       const adapter = await gpu.requestAdapter();
-      this.device = await adapter.requestDevice();
+      const holder: DeviceHolder = {
+        device: await adapter.requestDevice(),
+        lostReason: undefined,
+      };
+
+      holder.device.lost.then(ev => {
+        holder.lostReason = ev.message;
+      });
+
+      this.holder = holder;
+      this.state = 'free';
```

Comment:
IIUC, the holder is set to `undefined` when we discover on test finalization that the device has been lost, or we think there may have been an unrecoverable timeout. That means we should no longer try to use the device and instead make a new one on the next test run.

Comments in release() are helpful; I think it would help to add some comments about the device lifecycle here as well.